rules/Php74/Guard/PropertyTypeChangeGuard.php

```diff
@@ -67,24 +67,24 @@ public function isLegal(Property $property, bool $inlinePublic = true, bool $isC
             return true;
         }
 
-        return $this->isSafeProtectedProperty($property);
+        return $this->isSafeProtectedProperty($classReflection, $property);
     }
 
-    private function isSafeProtectedProperty(Property $property): bool
+    private function isSafeProtectedProperty(ClassReflection $classReflection, Property $property): bool
     {
         if (! $property->isProtected()) {
             return false;
         }
 
-        $parentNode = $property->getAttribute(AttributeKey::PARENT_NODE);
-        if (! $parentNode instanceof Class_) {
-            throw new ShouldNotHappenException();
+        $class = $this->classLikeAstResolver->resolveClassFromClassReflection($classReflection);
```

Comment:
It seems make unnecessary IO process,  I will check if it possibly to use Class_ node directly in the rule itself.

Reply:
https://github.com/rectorphp/rector-src/pull/3993